src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,1051 @@
+    std::vector<One_critical_filtration<U>> out(this->num_generators());
+    for (std::size_t i = 0u; i < this->num_generators(); ++i) {
+      out[i] = this->multi_filtration_[i].template astype<U>();
+    }
+    return Multi_critical_filtration<U>(std::move(out));
+  }
+  inline void push_back(const OneCritical& x){
+    multi_filtration_.push_back(x);
+  }
+  inline const std::vector<One_critical_filtration<T>>& as_vector() const {
+    return multi_filtration_;
+  }
+
+  inline std::vector<std::vector<T>> as_VECTOR() const {
+    std::vector<std::vector<T>> out(this->num_generators(),
+                                    std::vector<T>(this->num_parameters()));
+    for (std::size_t i = 0; i < this->num_generators(); ++i) {
+      for (std::size_t j = 0; j < this->num_parameters(); ++j) {
+        out[i][j] = multi_filtration_[i][j];
+      }
+    }
+    return out;
+  }
+
+  inline void _clean(bool keep_inf=true) {
+    multi_filtration_.erase(std::remove_if(multi_filtration_.begin(), multi_filtration_.end(),
+                              [keep_inf](const OneCritical &a) {
+                                return a.empty() ||
+                                       ((!keep_inf) &&
+                                        (a.is_inf() || a.is_minus_inf()));
+                              }),
+                            multi_filtration_.end());
+  }
+  inline std::size_t num_generators() const {
+    return multi_filtration_.size();
+  }
+
+
+
+  // TODO : this costs a lot... optimize / cheat in some way for python ?
+  /*
+  * Checks if `this`, seen as a birth curve is under the `other` birth curve,
+  *
+  */
+  inline bool operator<(const Multi_critical_filtration<T,co> &other) const {
+    //check if this curves is below other's curve 
+    // ie for each guy in this, check if there is a guy in other that dominates him
+    for (std::size_t i = 0u; i < multi_filtration_.size(); ++i) {
+      for (std::size_t j = 0u; j < other.multi_filtration_.size(); ++j) {
+        // i<j
+        if (strictly_dominates(other.multi_filtration_[j],multi_filtration_[i]))
+          continue;
+      }
+      return false;
+    }
+    return true;
+  }
+  /*
+  * Checks if `this`, seen as a birth curve is over the `other` birth curve,
+  */
+  inline bool operator>(const Multi_critical_filtration<T,co> &other) const {
+    return other < *this;
+  }
+
+  /*
+  * Checks if `this`, seen as a birth curve is under the `other` birth curve,
+  */
+  inline bool operator<=(const Multi_critical_filtration<T,co> &other) const {
+    //check if this curves is below other's curve 
+    // ie for each guy in this, check if there is a guy in other that dominates him
+    for (std::size_t i = 0u; i < multi_filtration_.size(); ++i) {
+      for (std::size_t j = 0u; j < other.multi_filtration_.size(); ++j) {
+        // i <= j 
+        if (dominates(other.multi_filtration_[j], multi_filtration_[i]))
+          continue;
```

Comment:
`continue` which loop? I think you want the outer one, but C++ understands the inner one.